dotcom-rendering/src/web/components/LightboxWrapper.importable.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+import { getIslandsByName } from '../browser/islands/getIslandsByName';
+import { getProps } from '../browser/islands/getProps';
+import { Lightbox } from './Lightbox';
+import type { Props as LightboxPictureProps } from './LightboxPicture';
+
+const isServer = typeof window === 'undefined';
+
+export type Props = LightboxPictureProps & {
+	caption?: string;
+};
+
+const getLightboxElements = (): Props[] => {
+	const elements = getIslandsByName('LightboxWrapper');
+	return elements.map((element) => getProps(element) as Props);
+};
+
+export const LightboxWrapper = (initialElementProps: Props) => {
+	const [isOpen, setOpen] = useState(true);
+	const [elements, setElements] = useState<Props[]>([]);
+	useEffect(() => {
+		setElements(getLightboxElements());
+	}, []);
```

Comment:
This is an interesting mix of plain js and react. Are there any edge cases to this approach that might cause issues?